addons/mail/static/src/discuss/call/common/call_invitation.js

```diff
@@ -9,20 +9,61 @@ export class CallInvitation extends Component {
     setup() {
         super.setup();
         this.rtc = useService("discuss.rtc");
+        this.ui = useState(useService("ui"));
+        this.state = useState({ videoStream: null });
+        this.videoRef = useRef("video");
+        onWillUnmount(() => {
+            if (!this.state.videoStream) {
+                return;
+            }
+            this.stopTracksOnMediaStream(this.state.videoStream);
+        });
     }
 
-    async onClickAccept(ev) {
+    async onClickAccept(ev, { camera = false } = {}) {
         this.props.thread.open();
         if (this.rtc.state.hasPendingRequest) {
             return;
         }
-        await this.rtc.toggleCall(this.props.thread);
+        await this.rtc.toggleCall(this.props.thread, { camera });
     }
 
     onClickAvatar(ev) {
         this.props.thread.open();
     }
 
+    get hasRtcSupport() {
+        return Boolean(
+            navigator.mediaDevices && navigator.mediaDevices.getUserMedia && window.MediaStream
+        );
+    }
```

Comment:
There are multiple occurrences of this code, we should centralize it somewhere (in master maybe)

Reply:
This is a change in 18.0, we should avoid changing accessibility of public methods, hence the copy-pasta

Reply:
But I agree this should be moved in store or rtc service